docs/madr/decisions/056-meshexternalservice-and-zoneegress.md

```diff
@@ -0,0 +1,371 @@
+
+- all the problems of running through egress are not applicable (operational cost, extra networking hop, scalability etc.)
+- no single exit point for all the external traffic
+- if we put credentials in the ExternalService then these credentials are accessible to the app owners if not delivered via generic_secret
+- it's hard to emulate ExternalService because there is no "from" section (there is no destination that is inside the mesh)
+
+#### Running traffic through sidecar and egress
+
+- we can mix and match the pros/cons of both approaches
+- it's harder to implement / explain how it's working
+
+### Configuration placement
+
+#### Current state and possible options
+
+##### Table summary
+
+The table below shows a summary of policies and if it makes sense to configure them on:
+- sidecar
+- egress
+- on both
+
+and if it makes sense for the user to pick where.
+
+We assume in this table that it never makes sense to put `from` section of a policy on the sidecar because there is no inbound for it, only an outbound (`to` section).
+Both `from` and `to` sections makes sense on the egress.
+
+| Policy                    | On sidecar    | On egress              | On both | User configurable |
+|---------------------------|---------------|------------------------|---------|-------------------|
+| MeshAccessLog             | Yes           | Yes                    | Yes     | Yes               |
+| MeshCircuitBreaker        | Yes           | Maybe                  | No      | No                |
+| MeshFaultInjection        | Maybe         | Yes                    | No      | No                |
+| MeshHealthCheck           | Yes           | Maybe                  | No      | No                |
+| MeshMetric                | Yes           | Yes                    | Yes     | Maybe             |
+| MeshProxyPatch            | Yes           | Yes                    | Yes     | Yes               |
+| MeshRateLimit             | Probably not  | Yes                    | No      | No                |
+| MeshRetry                 | Yes           | Maybe                  | No      | Maybe             |
+| MeshTimeout               | Yes           | Maybe                  | No      | No                |
+| MeshTrace                 | Yes           | Yes                    | Yes     | Maybe             |
+| MeshTrafficPermission     | Maybe         | Yes                    | Maybe   | Maybe             |
+| MeshLoadBalancingStrategy | Yes           | Yes                    | No      | Maybe             |
+| MeshTCPRoute              | Yes           | Yes                    | No      | No                |
+| MeshHTTPRoute             | Yes           | Yes                    | No      | No                |
+
+In paragraphs below I go deeper into each policy.
+
+##### MeshAccessLog
+
+MeshAccessLog is not applied on egress at all, and we don't provide a way to configure it.
+I don't see any reason not to do this though.
```

Comment:
I guess one issue is that zoneEgress is multi-mesh whereas other components usually are not.

Reply:
does multi-mesh play a role here?

Reply:
MeshAccessLog is configured per TCPProxy (or HTTPConnectionManager), so we can set
```
kind: MeshAccessLog
metadata:
  labels:
    kuma.io/mesh: mesh-1
spec:
  targetRef:
    kind: ZoneEgress
    name: zoneegress-1
```
only for FilterChains related to `mesh-1`.

Reply:
Added note about this.